tests/unit/test_analytics.py

```diff
@@ -88,7 +86,74 @@ def test_find_periods(
         (dummy_datetime_list[9], dummy_datetime_list[9]),
     ]
 
-    result = sleep_detection._find_periods(window_measurement)
+    result = analytics._find_periods(window_measurement)
+
+    assert result == expected_result, f"Expected {expected_result}, but got {result}"
+
+
+@pytest.mark.parametrize(
+    "non_wear_array, expected_result",
+    [
+        (np.array([0, 1, 1, 1, 0]), []),
+        (np.array([0, 0, 0, 1, 1]), []),
+        (np.array([0, 0, 1, 0, 0]), []),
+    ],
+)
+def test_remove_nonwear_periods_overlap(
+    non_wear_array: np.ndarray, expected_result: List
+) -> None:
+    """Test the _remove_nonwear_from_sleep method.
+
+    This test is for the following cases:
+        - where the non-wear period is at the beginning of the sleep window.
+        - where the non-wear period is at the end of the sleep window.
+        - where the non-wear period is within the sleep window.
+    """
+    dummy_date = datetime(2024, 5, 2)
+    sleep_windows = []
+    sleep_windows.append(
+        analytics.SleepWindow(
+            onset=dummy_date + timedelta(hours=2),
+            wakeup=dummy_date + timedelta(hours=4),
+        )
+    )
```

Comment:
this is likely superfluous but would it be worth adding just one (non overlapping) window just to "prove" it's not just throwing out everything when there's an overlap.

Reply:
Yeah, do you mean like in the `test_remove_nonwear_periods_no_overlap`? It is the test right after this one

Reply:
In my head I was thinking more of demonstrating the function was able to _selectively_ remove the bad sleep period while still retaining the good one, but thats just smooshing the two tests together now that I type it out, so go ahead and ignore this ha.